server/controllers/changeTemporaryDeactivationDetails/details.ts

```diff
@@ -0,0 +1,152 @@
+
+    deactivationReason.items = Object.entries(deactivationReasons)
+      .sort(([a, _], [b, __]) => {
+        if ([a, b].includes('OTHER')) {
+          return a === 'OTHER' ? 1 : -1
+        }
+
+        return a.localeCompare(b)
+      })
+      .map(([key, value]) => ({
+        text: value,
+        value: key,
+        conditional: `deactivationReason${key === 'OTHER' ? 'Other' : `Description-${key}`}`,
+      }))
+
+    Object.keys(deactivationReasons)
+      .filter(n => n !== 'OTHER')
+      .forEach(key => {
+        req.form.options.allFields[`deactivationReasonDescription-${key}`] = {
+          ...req.form.options.allFields.deactivationReasonDescription,
+          id: `deactivationReasonDescription-${key}`,
+          name: `deactivationReasonDescription-${key}`,
+        }
+      })
+
+    delete req.form.options.allFields.deactivationReasonDescription
+
+    next()
+  }
+
+  validateFields(req: FormWizard.Request, res: Response, callback: (errors: any) => void) {
+    req.form.values.deactivationReasonDescription =
+      req.body[`deactivationReasonDescription-${req.form.values.deactivationReason}`]
+    super.validateFields(req, res, callback)
+  }
+
+  locals(req: FormWizard.Request, res: Response): object {
+    const locals = super.locals(req, res)
+
+    const { id: locationId, prisonId } = res.locals.location
+
+    const backLink = backUrl(req, {
+      fallbackUrl: `/view-and-update-locations/${prisonId}/${locationId}`,
+      nextStepUrl: `/location/${locationId}/deactivate/temporary/confirm`,
+    })
+    return {
+      ...locals,
+      backLink,
+      cancelLink: `/view-and-update-locations/${prisonId}/${locationId}`,
+    }
+  }
+
+  compareInitialAndSubmittedValues(values: { initialValues: any; submittedValues: any }) {
+    return JSON.stringify(values.initialValues) !== JSON.stringify(values.submittedValues)
+  }
+
+  async saveValues(req: FormWizard.Request, res: Response, next: NextFunction) {
+    try {
+      const { user, location } = res.locals
+      const { locationsService } = req.services
+
+      const { deactivationReason, estimatedReactivationDate, planetFmReference } = req.form.values
+
+      const token = await req.services.authService.getSystemClientToken(user.username)
+      const deactivationReasonDesc = `deactivationReason${deactivationReason === 'OTHER' ? 'Other' : `Description-${deactivationReason}`}`
+      const submittedValues = {
+        deactivationReason: deactivationReason as string,
+        [deactivationReasonDesc]: req.form.values[
+          `deactivationReason${deactivationReason === 'OTHER' ? 'Other' : 'Description'}`
+        ] as string,
+
+        estimatedReactivationDate: estimatedReactivationDate as string,
+        planetFmReference: planetFmReference as string,
+      }
+
+      const valuesHaveChanged = this.compareInitialAndSubmittedValues({
+        initialValues: this.getInitialValues(req, res),
+        submittedValues,
+      })
```

Comment:
A lot of the other journeys have what I think is the same logic, where it checks if the values have changed and if not then it returns to the view location page, but it's implemented in a different way here:
https://github.com/ministryofjustice/hmpps-locations-inside-prison/blob/main/server/controllers/changeCellCapacity/index.ts#L49-L60

Not a big issue but something to be aware of maybe